functests/1_performance/netqueues.go

```diff
@@ -0,0 +1,359 @@
+					tunedPod := getTunedPod(&node)
+					out, err := pods.ExecCommandOnPod(tunedPod, tunedCmd)
+					deviceExists := strings.ContainsAny(string(out), device)
+					Expect(deviceExists).To(BeTrue())
+					Expect(err).ToNot(HaveOccurred())
+				}
+			}
+		})
+	})
+})
+
+//Check if the device support multiple queues
+func checkDeviceSupport(workernodes []corev1.Node, devices map[string]int) error {
+	cmdGetPhysicalDevices := []string{"find", "/sys/class/net", "-type", "l", "-not", "-lname", "*virtual*", "-printf", "%f "}
+	var channelCurrentCombined int
+	var err error
+	for _, node := range workernodes {
+		tunedPod := getTunedPod(&node)
+		phyDevs, err := pods.ExecCommandOnPod(tunedPod, cmdGetPhysicalDevices)
+		Expect(err).ToNot(HaveOccurred())
+		for _, d := range strings.Split(string(phyDevs), " ") {
+			if d == "" {
+				continue
+			}
+			_, err := pods.ExecCommandOnPod(tunedPod, []string{"ethtool", "-l", d})
+			if err == nil {
+				cmdCombinedChannelsCurrent := []string{"bash", "-c",
+					fmt.Sprintf("ethtool -l %s | sed -n '/Current hardware settings:/,/Combined:/{s/^Combined:\\s*//p}'", d)}
+				out, err := pods.ExecCommandOnPod(tunedPod, cmdCombinedChannelsCurrent)
+				if strings.Contains(string(out), "n/a") {
+					fmt.Printf("Device %s doesn't support multiple queues\n", d)
+				} else {
+					channelCurrentCombined, err = strconv.Atoi(strings.TrimSpace(string(out)))
+					if err != nil {
+						testlog.Warningf(fmt.Sprintf("unable to retrieve current multi-purpose channels hardware settings for device %s on %s",
+							d, node.Name))
+					}
+					if channelCurrentCombined == 1 {
+						fmt.Printf("Device %s doesn't support multiple queues\n", d)
+					} else {
+						fmt.Printf("Device %s supports multiple queues\n", d)
+						devices[d] = channelCurrentCombined
+					}
+				}
+			}
+		}
+	}
+	// check the length of the map, if the map is empty , then
+	// there are no supported devices. Hence we skip
+	if len(devices) == 0 {
+		Skip("Skipping Test: There are no supported Network Devices")
+	}
+	return err
+}
+
+//Get Tuned Pods
+func getTunedPod(node *corev1.Node) *corev1.Pod {
+	listOptions := &client.ListOptions{
+		Namespace:     components.NamespaceNodeTuningOperator,
+		FieldSelector: fields.SelectorFromSet(fields.Set{"spec.nodeName": node.Name}),
+		LabelSelector: labels.SelectorFromSet(labels.Set{"openshift-app": "tuned"}),
+	}
+	tunedList := &corev1.PodList{}
+	Eventually(func() bool {
+		if err := testclient.Client.List(context.TODO(), tunedList, listOptions); err != nil {
+			return false
+		}
+		if len(tunedList.Items) == 0 {
+			return false
+		}
+		for _, s := range tunedList.Items[0].Status.ContainerStatuses {
+			if s.Ready == false {
+				return false
+			}
+		}
+		return true
+	}, cluster.ComputeTestTimeout(testTimeout, RunningOnSingleNode), testPollInterval*time.Second).Should(BeTrue(),
+		"there should be one tuned daemon per node")
+	return &tunedList.Items[0]
+}
+
+//Get Reserved CPU Count
```

Comment:
when the functions are not public (e.g start with capitals) I dont see the need for the comment above if the name is clear

Reply:
okay addressed it in the latest patch.